### PR TITLE
fix shape tests for broadcast op

### PR DESCRIPTION
**Pull request by kahmed10**

### CODE REVIEW

File: src/include/migraphx/op/broadcast.hpp [hunk 1]
```diff
@@ -46,25 +46,21 @@ struct broadcast
...
-            return {t, broadcast_lens, std::move(bcast_strides)};
+        if(!std::equal(input.lens().begin(), input.lens().end(), broadcast_lens.begin() + axis))
+        {
+            MIGRAPHX_THROW("BROADCAST: when broadcasting success sizes must match");
         }
+        std::copy(input.strides().begin(), input.strides().end(), bcast_strides.begin() + axis);
+
+        shape output{t, broadcast_lens, std::move(bcast_strides)};
+        if(output.elements() < input.elements())
+            MIGRAPHX_THROW("BROADCAST: output size must be greater than or equal to input size");
```
Comment (pfultz2) [inline src/include/migraphx/op/broadcast.hpp:62]: It seems you need another case to get coverage for this branch.

Comment (kahmed10) [reply src/include/migraphx/op/broadcast.hpp:62]: fixed, and added another test for the other missing codecov case

File: src/include/migraphx/op/broadcast.hpp [hunk 2]
```diff
@@ -46,25 +46,21 @@ struct broadcast
             MIGRAPHX_THROW("BROADCAST : axis is out of range");
         }
 
-        if(std::all_of(
-               broadcast_lens.cbegin(), broadcast_lens.cend(), [&](auto x) { return x == 1; }))
+        if(broadcast_lens.size() - axis < input.lens().size())
         {
-            return {t, broadcast_lens, std::move(bcast_strides)};
+            MIGRAPHX_THROW("BROADCAST: when broadcasting success sizes must match");
```
Comment (scxiao) [inline src/include/migraphx/op/broadcast.hpp:51]: I think the error message is incorrect here

File: src/include/migraphx/op/broadcast.hpp [hunk 3]
```diff
@@ -46,25 +46,21 @@ struct broadcast
...
 
-            if(!std::equal(input.lens().begin(), input.lens().end(), broadcast_lens.begin() + axis))
-            {
-                MIGRAPHX_THROW("BROADCAST: when broadcasting success sizes must match");
-            }
-            std::copy(input.strides().begin(), input.strides().end(), bcast_strides.begin() + axis);
-            return {t, broadcast_lens, std::move(bcast_strides)};
+        if(!std::equal(input.lens().begin(), input.lens().end(), broadcast_lens.begin() + axis))
+        {
+            MIGRAPHX_THROW("BROADCAST: when broadcasting success sizes must match");
```
Comment (scxiao) [inline src/include/migraphx/op/broadcast.hpp:56]: success --> succeeding?

**Review by pfultz2 — APPROVED** [review]

File: src/include/migraphx/op/broadcast.hpp (hunk 3 above)
Comment (kahmed10) [reply src/include/migraphx/op/broadcast.hpp]: fixed

File: src/include/migraphx/op/broadcast.hpp (hunk 2 above)
Comment (kahmed10) [reply src/include/migraphx/op/broadcast.hpp]: fixed

**Review by scxiao — APPROVED** [review]

**Pull request merged**